ClassCastException with Gradle 7.5-milestone-1

#### I'm submitting a ... 
  - [x] bug report
  - [ ] feature request
  - [ ] support request => Please do not submit support request here, see note at the top of this template.

#### What is the current behavior?
After upgrading from Gradle `7.4.2` to `7.5-milestone-1` I get a stacktrace (see below).

#### If the current behavior is a bug, please provide the steps to reproduce and if possible a minimal demo of the problem
Invoke `gradle allureServe` with Gradle `7.5-milestone-1`

#### What is the expected behavior?
No stacktrace as before.

#### What is the motivation / use case for changing the behavior?
Using Allure as before

#### Please tell us about your environment:

| Allure version | 2.17.3 |
| --- | --- |
| Test framework | JUnit 5.8.2 |
| Allure adaptor | allure-junit 2.17.3 |
| Generate report using | Gradle Plugin 2.9.6 |

#### Other information 

```
> gradle allureServe --info --stacktrace
...
Task name matched 'allureServe'
Selected primary task 'allureServe' from project :

FAILURE: Build failed with an exception.

* What went wrong:
class org.gradle.execution.plan.ActionNode cannot be cast to class org.gradle.execution.plan.TaskNode (org.gradle.execution.plan.ActionNode and org.gradle.execution.plan.TaskNode are in unnamed module of loader org.gradle.internal.classloader.VisitableURLClassLoader @1c4af82c)

...

* Exception is:
java.lang.ClassCastException: class org.gradle.execution.plan.ActionNode cannot be cast to class org.gradle.execution.plan.TaskNode (org.gradle.execution.plan.ActionNode and org.gradle.execution.plan.TaskNode are in unnamed module of loader org.gradle.internal.classloader.VisitableURLClassLoader @1c4af82c)
        at org.gradle.execution.plan.LocalTaskNode.resolveDependencies(LocalTaskNode.java:148)
        at org.gradle.execution.plan.DefaultExecutionPlan.doAddNodes(DefaultExecutionPlan.java:209)
        at org.gradle.execution.plan.DefaultExecutionPlan.addEntryTasks(DefaultExecutionPlan.java:174)
        at org.gradle.execution.plan.DefaultExecutionPlan.addEntryTasks(DefaultExecutionPlan.java:159)
        at org.gradle.execution.TaskNameResolvingBuildConfigurationAction.configure(TaskNameResolvingBuildConfigurationAction.java:47)
        at org.gradle.execution.DefaultBuildConfigurationActionExecuter.configure(DefaultBuildConfigurationActionExecuter.java:49)
        at org.gradle.execution.DefaultBuildConfigurationActionExecuter.access$000(DefaultBuildConfigurationActionExecuter.java:25)
        at org.gradle.execution.DefaultBuildConfigurationActionExecuter$1.proceed(DefaultBuildConfigurationActionExecuter.java:62)
        at org.gradle.execution.DefaultTasksBuildExecutionAction.configure(DefaultTasksBuildExecutionAction.java:48)
        at org.gradle.execution.DefaultBuildConfigurationActionExecuter.configure(DefaultBuildConfigurationActionExecuter.java:49)
        at org.gradle.execution.DefaultBuildConfigurationActionExecuter.lambda$select$0(DefaultBuildConfigurationActionExecuter.java:36)
        at org.gradle.internal.Factories$1.create(Factories.java:31)
        at org.gradle.internal.work.DefaultWorkerLeaseService.withReplacedLocks(DefaultWorkerLeaseService.java:345)
        at org.gradle.api.internal.project.DefaultProjectStateRegistry$DefaultBuildProjectRegistry.withMutableStateOfAllProjects(DefaultProjectStateRegistry.java:197)
        at org.gradle.api.internal.project.DefaultProjectStateRegistry$DefaultBuildProjectRegistry.withMutableStateOfAllProjects(DefaultProjectStateRegistry.java:190)
        at org.gradle.execution.DefaultBuildConfigurationActionExecuter.select(DefaultBuildConfigurationActionExecuter.java:35)
        at org.gradle.initialization.DefaultTaskExecutionPreparer.prepareForTaskExecution(DefaultTaskExecutionPreparer.java:42)
        at org.gradle.initialization.VintageBuildModelController.lambda$scheduleRequestedTasks$1(VintageBuildModelController.java:81)
        at org.gradle.internal.model.StateTransitionController.lambda$inState$1(StateTransitionController.java:110)
        at org.gradle.internal.model.StateTransitionController.lambda$inState$2(StateTransitionController.java:125)
        at org.gradle.internal.work.DefaultSynchronizer.withLock(DefaultSynchronizer.java:44)
        at org.gradle.internal.model.StateTransitionController.inState(StateTransitionController.java:121)
        at org.gradle.internal.model.StateTransitionController.inState(StateTransitionController.java:109)
        at org.gradle.initialization.VintageBuildModelController.scheduleRequestedTasks(VintageBuildModelController.java:81)
        at org.gradle.internal.build.DefaultBuildLifecycleController$DefaultWorkGraphBuilder.addRequestedTasks(DefaultBuildLifecycleController.java:242)
        at org.gradle.internal.build.DefaultBuildLifecycleController.lambda$populateWorkGraph$4(DefaultBuildLifecycleController.java:143)
        at org.gradle.internal.build.DefaultBuildWorkPreparer.populateWorkGraph(DefaultBuildWorkPreparer.java:41)
        at org.gradle.internal.build.BuildOperationFiringBuildWorkPreparer$PopulateWorkGraph.populateTaskGraph(BuildOperationFiringBuildWorkPreparer.java:138)
        at org.gradle.internal.build.BuildOperationFiringBuildWorkPreparer$PopulateWorkGraph.run(BuildOperationFiringBuildWorkPreparer.java:89)
        at org.gradle.internal.operations.DefaultBuildOperationRunner$1.execute(DefaultBuildOperationRunner.java:29)
        at org.gradle.internal.operations.DefaultBuildOperationRunner$1.execute(DefaultBuildOperationRunner.java:26)
        at org.gradle.internal.operations.DefaultBuildOperationRunner$2.execute(DefaultBuildOperationRunner.java:66)
        at org.gradle.internal.operations.DefaultBuildOperationRunner$2.execute(DefaultBuildOperationRunner.java:59)
        at org.gradle.internal.operations.DefaultBuildOperationRunner.execute(DefaultBuildOperationRunner.java:157)
        at org.gradle.internal.operations.DefaultBuildOperationRunner.execute(DefaultBuildOperationRunner.java:59)
        at org.gradle.internal.operations.DefaultBuildOperationRunner.run(DefaultBuildOperationRunner.java:47)
        at org.gradle.internal.operations.DefaultBuildOperationExecutor.run(DefaultBuildOperationExecutor.java:68)
        at org.gradle.internal.build.BuildOperationFiringBuildWorkPreparer.populateWorkGraph(BuildOperationFiringBuildWorkPreparer.java:66)
        at org.gradle.internal.build.DefaultBuildLifecycleController.lambda$populateWorkGraph$5(DefaultBuildLifecycleController.java:143)
        at org.gradle.internal.model.StateTransitionController.lambda$inState$1(StateTransitionController.java:110)
        at org.gradle.internal.model.StateTransitionController.lambda$inState$2(StateTransitionController.java:125)
        at org.gradle.internal.work.DefaultSynchronizer.withLock(DefaultSynchronizer.java:44)
        at org.gradle.internal.model.StateTransitionController.inState(StateTransitionController.java:121)
        at org.gradle.internal.model.StateTransitionController.inState(StateTransitionController.java:109)
        at org.gradle.internal.build.DefaultBuildLifecycleController.populateWorkGraph(DefaultBuildLifecycleController.java:143)
        at org.gradle.internal.build.DefaultBuildWorkGraphController$DefaultBuildWorkGraph.populateWorkGraph(DefaultBuildWorkGraphController.java:126)
        at org.gradle.composite.internal.DefaultBuildController.populateWorkGraph(DefaultBuildController.java:71)
        at org.gradle.composite.internal.DefaultIncludedBuildTaskGraph$DefaultBuildTreeWorkGraphBuilder.withWorkGraph(DefaultIncludedBuildTaskGraph.java:141)
        at org.gradle.internal.buildtree.DefaultBuildTreeWorkPreparer.lambda$scheduleRequestedTasks$0(DefaultBuildTreeWorkPreparer.java:34)
        at org.gradle.composite.internal.DefaultIncludedBuildTaskGraph$DefaultBuildTreeWorkGraph$1.run(DefaultIncludedBuildTaskGraph.java:169)
        at org.gradle.internal.operations.DefaultBuildOperationRunner$1.execute(DefaultBuildOperationRunner.java:29)
        at org.gradle.internal.operations.DefaultBuildOperationRunner$1.execute(DefaultBuildOperationRunner.java:26)
        at org.gradle.internal.operations.DefaultBuildOperationRunner$2.execute(DefaultBuildOperationRunner.java:66)
        at org.gradle.internal.operations.DefaultBuildOperationRunner$2.execute(DefaultBuildOperationRunner.java:59)
        at org.gradle.internal.operations.DefaultBuildOperationRunner.execute(DefaultBuildOperationRunner.java:157)
        at org.gradle.internal.operations.DefaultBuildOperationRunner.execute(DefaultBuildOperationRunner.java:59)
        at org.gradle.internal.operations.DefaultBuildOperationRunner.run(DefaultBuildOperationRunner.java:47)
        at org.gradle.internal.operations.DefaultBuildOperationExecutor.run(DefaultBuildOperationExecutor.java:68)
        at org.gradle.composite.internal.DefaultIncludedBuildTaskGraph$DefaultBuildTreeWorkGraph.scheduleWork(DefaultIncludedBuildTaskGraph.java:166)
        at org.gradle.internal.buildtree.DefaultBuildTreeWorkPreparer.scheduleRequestedTasks(DefaultBuildTreeWorkPreparer.java:34)
        at org.gradle.internal.buildtree.DefaultBuildTreeLifecycleController.lambda$doScheduleAndRunTasks$2(DefaultBuildTreeLifecycleController.java:89)
        at org.gradle.composite.internal.DefaultIncludedBuildTaskGraph.withNewWorkGraph(DefaultIncludedBuildTaskGraph.java:100)
        at org.gradle.internal.buildtree.DefaultBuildTreeLifecycleController.doScheduleAndRunTasks(DefaultBuildTreeLifecycleController.java:88)
        at org.gradle.internal.buildtree.DefaultBuildTreeLifecycleController.lambda$runBuild$4(DefaultBuildTreeLifecycleController.java:106)
        at org.gradle.internal.model.StateTransitionController.lambda$transition$5(StateTransitionController.java:166)
        at org.gradle.internal.model.StateTransitionController.doTransition(StateTransitionController.java:247)
        at org.gradle.internal.model.StateTransitionController.lambda$transition$6(StateTransitionController.java:166)
        at org.gradle.internal.work.DefaultSynchronizer.withLock(DefaultSynchronizer.java:44)
        at org.gradle.internal.model.StateTransitionController.transition(StateTransitionController.java:166)
        at org.gradle.internal.buildtree.DefaultBuildTreeLifecycleController.runBuild(DefaultBuildTreeLifecycleController.java:103)
        at org.gradle.internal.buildtree.DefaultBuildTreeLifecycleController.scheduleAndRunTasks(DefaultBuildTreeLifecycleController.java:69)
        at org.gradle.tooling.internal.provider.ExecuteBuildActionRunner.run(ExecuteBuildActionRunner.java:31)
        at org.gradle.launcher.exec.ChainingBuildActionRunner.run(ChainingBuildActionRunner.java:35)
        at org.gradle.internal.buildtree.ProblemReportingBuildActionRunner.run(ProblemReportingBuildActionRunner.java:49)
        at org.gradle.launcher.exec.BuildOutcomeReportingBuildActionRunner.run(BuildOutcomeReportingBuildActionRunner.java:65)
        at org.gradle.tooling.internal.provider.FileSystemWatchingBuildActionRunner.run(FileSystemWatchingBuildActionRunner.java:136)
        at org.gradle.launcher.exec.BuildCompletionNotifyingBuildActionRunner.run(BuildCompletionNotifyingBuildActionRunner.java:41)
        at org.gradle.launcher.exec.RootBuildLifecycleBuildActionExecutor.lambda$execute$0(RootBuildLifecycleBuildActionExecutor.java:40)
        at org.gradle.composite.internal.DefaultRootBuildState.run(DefaultRootBuildState.java:122)
        at org.gradle.launcher.exec.RootBuildLifecycleBuildActionExecutor.execute(RootBuildLifecycleBuildActionExecutor.java:40)
        at org.gradle.internal.buildtree.DefaultBuildTreeContext.execute(DefaultBuildTreeContext.java:40)
        at org.gradle.launcher.exec.BuildTreeLifecycleBuildActionExecutor.lambda$execute$0(BuildTreeLifecycleBuildActionExecutor.java:65)
        at org.gradle.internal.buildtree.BuildTreeState.run(BuildTreeState.java:53)
        at org.gradle.launcher.exec.BuildTreeLifecycleBuildActionExecutor.execute(BuildTreeLifecycleBuildActionExecutor.java:65)
        at org.gradle.launcher.exec.RunAsBuildOperationBuildActionExecutor$3.call(RunAsBuildOperationBuildActionExecutor.java:61)
        at org.gradle.launcher.exec.RunAsBuildOperationBuildActionExecutor$3.call(RunAsBuildOperationBuildActionExecutor.java:57)
        at org.gradle.internal.operations.DefaultBuildOperationRunner$CallableBuildOperationWorker.execute(DefaultBuildOperationRunner.java:204)
        at org.gradle.internal.operations.DefaultBuildOperationRunner$CallableBuildOperationWorker.execute(DefaultBuildOperationRunner.java:199)
        at org.gradle.internal.operations.DefaultBuildOperationRunner$2.execute(DefaultBuildOperationRunner.java:66)
        at org.gradle.internal.operations.DefaultBuildOperationRunner$2.execute(DefaultBuildOperationRunner.java:59)
        at org.gradle.internal.operations.DefaultBuildOperationRunner.execute(DefaultBuildOperationRunner.java:157)
        at org.gradle.internal.operations.DefaultBuildOperationRunner.execute(DefaultBuildOperationRunner.java:59)
        at org.gradle.internal.operations.DefaultBuildOperationRunner.call(DefaultBuildOperationRunner.java:53)
        at org.gradle.internal.operations.DefaultBuildOperationExecutor.call(DefaultBuildOperationExecutor.java:73)
        at org.gradle.launcher.exec.RunAsBuildOperationBuildActionExecutor.execute(RunAsBuildOperationBuildActionExecutor.java:57)
        at org.gradle.launcher.exec.RunAsWorkerThreadBuildActionExecutor.lambda$execute$0(RunAsWorkerThreadBuildActionExecutor.java:36)
        at org.gradle.internal.work.DefaultWorkerLeaseService.withLocks(DefaultWorkerLeaseService.java:249)
        at org.gradle.internal.work.DefaultWorkerLeaseService.runAsWorkerThread(DefaultWorkerLeaseService.java:109)
        at org.gradle.launcher.exec.RunAsWorkerThreadBuildActionExecutor.execute(RunAsWorkerThreadBuildActionExecutor.java:36)
        at org.gradle.tooling.internal.provider.continuous.ContinuousBuildActionExecutor.execute(ContinuousBuildActionExecutor.java:110)
        at org.gradle.tooling.internal.provider.SubscribableBuildActionExecutor.execute(SubscribableBuildActionExecutor.java:64)
        at org.gradle.internal.session.DefaultBuildSessionContext.execute(DefaultBuildSessionContext.java:46)
        at org.gradle.tooling.internal.provider.BuildSessionLifecycleBuildActionExecuter$ActionImpl.apply(BuildSessionLifecycleBuildActionExecuter.java:100)
        at org.gradle.tooling.internal.provider.BuildSessionLifecycleBuildActionExecuter$ActionImpl.apply(BuildSessionLifecycleBuildActionExecuter.java:88)
        at org.gradle.internal.session.BuildSessionState.run(BuildSessionState.java:69)
        at org.gradle.tooling.internal.provider.BuildSessionLifecycleBuildActionExecuter.execute(BuildSessionLifecycleBuildActionExecuter.java:62)
        at org.gradle.tooling.internal.provider.BuildSessionLifecycleBuildActionExecuter.execute(BuildSessionLifecycleBuildActionExecuter.java:41)
        at org.gradle.tooling.internal.provider.StartParamsValidatingActionExecuter.execute(StartParamsValidatingActionExecuter.java:63)
        at org.gradle.tooling.internal.provider.StartParamsValidatingActionExecuter.execute(StartParamsValidatingActionExecuter.java:31)
        at org.gradle.tooling.internal.provider.SessionFailureReportingActionExecuter.execute(SessionFailureReportingActionExecuter.java:52)
        at org.gradle.tooling.internal.provider.SessionFailureReportingActionExecuter.execute(SessionFailureReportingActionExecuter.java:40)
        at org.gradle.tooling.internal.provider.SetupLoggingActionExecuter.execute(SetupLoggingActionExecuter.java:47)
        at org.gradle.tooling.internal.provider.SetupLoggingActionExecuter.execute(SetupLoggingActionExecuter.java:31)
        at org.gradle.launcher.daemon.server.exec.ExecuteBuild.doBuild(ExecuteBuild.java:65)
        at org.gradle.launcher.daemon.server.exec.BuildCommandOnly.execute(BuildCommandOnly.java:37)
        at org.gradle.launcher.daemon.server.api.DaemonCommandExecution.proceed(DaemonCommandExecution.java:104)
        at org.gradle.launcher.daemon.server.exec.WatchForDisconnection.execute(WatchForDisconnection.java:39)
        at org.gradle.launcher.daemon.server.api.DaemonCommandExecution.proceed(DaemonCommandExecution.java:104)
        at org.gradle.launcher.daemon.server.exec.ResetDeprecationLogger.execute(ResetDeprecationLogger.java:29)
        at org.gradle.launcher.daemon.server.api.DaemonCommandExecution.proceed(DaemonCommandExecution.java:104)
        at org.gradle.launcher.daemon.server.exec.RequestStopIfSingleUsedDaemon.execute(RequestStopIfSingleUsedDaemon.java:35)
        at org.gradle.launcher.daemon.server.api.DaemonCommandExecution.proceed(DaemonCommandExecution.java:104)
        at org.gradle.launcher.daemon.server.exec.ForwardClientInput$2.create(ForwardClientInput.java:78)
        at org.gradle.launcher.daemon.server.exec.ForwardClientInput$2.create(ForwardClientInput.java:75)
        at org.gradle.util.internal.Swapper.swap(Swapper.java:38)
        at org.gradle.launcher.daemon.server.exec.ForwardClientInput.execute(ForwardClientInput.java:75)
        at org.gradle.launcher.daemon.server.api.DaemonCommandExecution.proceed(DaemonCommandExecution.java:104)
        at org.gradle.launcher.daemon.server.exec.LogAndCheckHealth.execute(LogAndCheckHealth.java:55)
        at org.gradle.launcher.daemon.server.api.DaemonCommandExecution.proceed(DaemonCommandExecution.java:104)
        at org.gradle.launcher.daemon.server.exec.LogToClient.doBuild(LogToClient.java:63)
        at org.gradle.launcher.daemon.server.exec.BuildCommandOnly.execute(BuildCommandOnly.java:37)
        at org.gradle.launcher.daemon.server.api.DaemonCommandExecution.proceed(DaemonCommandExecution.java:104)
        at org.gradle.launcher.daemon.server.exec.EstablishBuildEnvironment.doBuild(EstablishBuildEnvironment.java:84)
        at org.gradle.launcher.daemon.server.exec.BuildCommandOnly.execute(BuildCommandOnly.java:37)
        at org.gradle.launcher.daemon.server.api.DaemonCommandExecution.proceed(DaemonCommandExecution.java:104)
        at org.gradle.launcher.daemon.server.exec.StartBuildOrRespondWithBusy$1.run(StartBuildOrRespondWithBusy.java:52)
        at org.gradle.launcher.daemon.server.DaemonStateCoordinator$1.run(DaemonStateCoordinator.java:297)
        at org.gradle.internal.concurrent.ExecutorPolicy$CatchAndRecordFailures.onExecute(ExecutorPolicy.java:64)
        at org.gradle.internal.concurrent.ManagedExecutorImpl$1.run(ManagedExecutorImpl.java:48)
```

--- a/.github/workflows/build.yml
+++ b/.github/workflows/build.yml
@@ -18,6 +18,11 @@ jobs:
       matrix:
         os: [ 'ubuntu', 'windows' ]
         java-version: [ '8.0.x', '11.0.x' ]
+        exclude:
+          - os: windows
+            # Java 8 + Windows cause build crashes for unknown reason
+            # https://github.com/gradle/gradle/issues/3093 might be relevant
+            java-version: '8.0.x'
     steps:
       - uses: actions/checkout@v2
 

--- a/allure-adapter-plugin/src/test/kotlin/io/qameta/allure/gradle/adapter/AdaptersTest.kt
+++ b/allure-adapter-plugin/src/test/kotlin/io/qameta/allure/gradle/adapter/AdaptersTest.kt
@@ -34,6 +34,12 @@ class AdaptersTest {
         @Parameterized.Parameters(name = "{1} [{0}]")
         fun getFrameworks() = listOf(
             arrayOf(
+                "7.5.1",
+                "src/it/adapter-junit5-spock-kts",
+                arrayOf("printAdapters"),
+                "[AdapterConfig{junit5}, AdapterConfig{spock}]"
+            ),
+            arrayOf(
                 "7.0",
                 "src/it/adapter-junit5-spock-kts",
                 arrayOf("printAdapters"),
@@ -46,6 +52,11 @@ class AdaptersTest {
                 "[AdapterConfig{junit5}, AdapterConfig{spock}]"
             ),
             arrayOf(
+                "7.5.1",
+                "src/it/adapter-all",
+                arrayOf("printAdapters"),
+                "[AdapterConfig{cucumber2Jvm}, AdapterConfig{cucumber3Jvm}, AdapterConfig{cucumber4Jvm}, AdapterConfig{cucumber5Jvm}, AdapterConfig{cucumber6Jvm}, AdapterConfig{cucumberJvm}, AdapterConfig{junit4}, AdapterConfig{junit5}, AdapterConfig{spock}, AdapterConfig{testng}]"
+            ),            arrayOf(
                 "7.0",
                 "src/it/adapter-all",
                 arrayOf("printAdapters"),

--- a/allure-base-plugin/src/main/kotlin/io/qameta/allure/gradle/base/tasks/AllureExecTask.kt
+++ b/allure-base-plugin/src/main/kotlin/io/qameta/allure/gradle/base/tasks/AllureExecTask.kt
@@ -9,6 +9,7 @@ import org.gradle.api.model.ObjectFactory
 import org.gradle.api.tasks.*
 import org.gradle.api.tasks.options.Option
 import org.gradle.kotlin.dsl.property
+import org.gradle.util.GradleVersion
 import java.io.File
 
 abstract class AllureExecTask constructor(objects: ObjectFactory) : DefaultTask() {
@@ -68,6 +69,12 @@ abstract class AllureExecTask constructor(objects: ObjectFactory) : DefaultTask(
         dependsOn(dependsOnTests.map { if (it) resultsDirs else emptyList<Any>() })
         // In any case, if user launches "./gradlew test allureReport" the report generation
         // should wait for test execution
-        mustRunAfter(resultsDirs)
+        if (GradleVersion.current() < GradleVersion.version("7.5")) {
+            mustRunAfter(resultsDirs)
+        } else {
+            // See https://github.com/allure-framework/allure-gradle/issues/90
+            // See https://github.com/gradle/gradle/issues/21962
+            mustRunAfter(resultsDirs.map { it.elements })
+        }
     }
 }

--- a/allure-plugin/src/test/java/io/qameta/allure/gradle/allure/CategoriesTest.java
+++ b/allure-plugin/src/test/java/io/qameta/allure/gradle/allure/CategoriesTest.java
@@ -38,6 +38,7 @@ public class CategoriesTest {
     @Parameterized.Parameters(name = "{1} [{0}]")
     public static Collection<Object[]> getFrameworks() {
         return Arrays.asList(
+                new Object[]{"7.5.1", "src/it/categories", new String[]{"allureReport"}},
                 new Object[]{"7.0", "src/it/categories", new String[]{"allureReport"}},
                 new Object[]{"5.0", "src/it/categories", new String[]{"allureReport"}},
                 new Object[]{"6.0", "src/it/categories", new String[]{"allureReport"}}

--- a/allure-plugin/src/test/java/io/qameta/allure/gradle/allure/DependenciesTest.java
+++ b/allure-plugin/src/test/java/io/qameta/allure/gradle/allure/DependenciesTest.java
@@ -28,16 +28,16 @@ public class DependenciesTest {
 
     // The order of versions is newest, oldest, rest
     private static final String[][] IT_MATRIX = {
-        { "src/it/cucumber-jvm",         "7.0", "5.0", "6.0" },
-        { "src/it/cucumber2-jvm",        "7.0", "5.0", "6.0" },
-        { "src/it/junit4",               "7.0", "5.0", "6.0" },
-        { "src/it/junit4-autoconfigure", "7.0", "5.0", "6.0" },
-        { "src/it/junit4-kotlin",        "7.0", "5.1", "5.0" },
-        { "src/it/junit5",               "7.0", "5.0", "6.0" },
-        { "src/it/junit5-5.8.1",         "7.2", "5.0", "6.0" },
-        { "src/it/testng",               "7.0", "5.0", "6.0" },
-        { "src/it/testng-autoconfigure", "7.0", "5.0", "6.0" },
-        { "src/it/spock",                "7.0", "5.0", "6.0" },
+        { "src/it/cucumber-jvm",         "7.5.1", "7.0", "5.0", "6.0" },
+        { "src/it/cucumber2-jvm",        "7.5.1", "7.0", "5.0", "6.0" },
+        { "src/it/junit4",               "7.5.1", "7.0", "5.0", "6.0" },
+        { "src/it/junit4-autoconfigure", "7.5.1", "7.0", "5.0", "6.0" },
+        { "src/it/junit4-kotlin",        "7.5.1", "7.0", "5.1", "5.0" },
+        { "src/it/junit5",               "7.5.1", "7.0", "5.0", "6.0" },
+        { "src/it/junit5-5.8.1",         "7.5.1", "7.2", "5.0", "6.0" },
+        { "src/it/testng",               "7.5.1", "7.0", "5.0", "6.0" },
+        { "src/it/testng-autoconfigure", "7.5.1", "7.0", "5.0", "6.0" },
+        { "src/it/spock",                "7.5.1", "7.0", "5.0", "6.0" },
     };
 
     @Parameterized.Parameter(0)

--- a/allure-plugin/src/test/java/io/qameta/allure/gradle/allure/TestNgSpiOffTest.java
+++ b/allure-plugin/src/test/java/io/qameta/allure/gradle/allure/TestNgSpiOffTest.java
@@ -31,7 +31,7 @@ public class TestNgSpiOffTest {
 
     @Parameterized.Parameters(name = "{0}")
     public static Collection<String> getFrameworks() {
-        return Arrays.asList("7.0", "6.0", "5.4", "5.3");
+        return Arrays.asList("7.5.1", "7.0", "6.0", "5.4", "5.3");
     }
 
     @Test

--- a/allure-plugin/src/test/kotlin/io/qameta/allure/gradle/report/DslTest.kt
+++ b/allure-plugin/src/test/kotlin/io/qameta/allure/gradle/report/DslTest.kt
@@ -28,9 +28,11 @@ class DslTest {
         @JvmStatic
         @Parameterized.Parameters(name = "{1} [{0}]")
         fun getFrameworks() = listOf(
+            arrayOf("7.5.1", "src/it/full-dsl-kotlin"),
             arrayOf("7.0", "src/it/full-dsl-kotlin"),
             arrayOf("6.0", "src/it/full-dsl-kotlin"),
             arrayOf("5.0", "src/it/full-dsl-kotlin"),
+            arrayOf("7.5.1", "src/it/full-dsl-groovy"),
             arrayOf("7.0", "src/it/full-dsl-groovy"),
             arrayOf("6.0", "src/it/full-dsl-groovy"),
             arrayOf("5.0", "src/it/full-dsl-groovy")

--- a/allure-plugin/src/test/kotlin/io/qameta/allure/gradle/report/TestAndAllureReportTest.kt
+++ b/allure-plugin/src/test/kotlin/io/qameta/allure/gradle/report/TestAndAllureReportTest.kt
@@ -38,7 +38,7 @@ class TestAndAllureReportTest {
         @Parameterized.Parameters(name = "{1},  [{0}]")
         fun getFrameworks(): Array<Any> {
             val res = mutableListOf<Array<Any>>()
-            for (gradleVersion in listOf("7.2", "7.0", "6.8.3")) {
+            for (gradleVersion in listOf("7.5.1", "7.2", "7.0", "6.8.3")) {
                 for (project in listOf("src/it/junit5-5.8.1")) {
                     for (dependOnTests in listOf(true, false)) {
                         res.add(arrayOf(gradleVersion, project, dependOnTests))

--- a/allure-report-plugin/src/test/java/io/qameta/allure/gradle/report/ReportOnlyTest.java
+++ b/allure-report-plugin/src/test/java/io/qameta/allure/gradle/report/ReportOnlyTest.java
@@ -31,6 +31,7 @@ public class ReportOnlyTest {
     @Parameterized.Parameters(name = "{0} [{1}]")
     public static Collection<Object[]> getFrameworks() {
         return Arrays.asList(
+                new Object[]{"src/it/report-only", "7.5.1"},
                 new Object[]{"src/it/report-only", "7.0"},
                 new Object[]{"src/it/report-only", "5.0"},
                 new Object[]{"src/it/report-only", "6.0"}

